chains/evm/contracts/onRamp/OnRamp.sol

```diff
@@ -222,10 +222,10 @@ contract OnRamp is IEVM2AnyOnRampClient, ITypeAndVersion, Ownable2StepMsgSender
       ccipReceiveGasLimit: resolvedExtraArgs.gasLimit,
       finality: resolvedExtraArgs.blockConfirmations,
       ccvAndExecutorHash: bytes32(0), // Will be set after CCV list is finalized.
-      onRampAddress: abi.encodePacked(address(this)),
+      onRampAddress: abi.encode(address(this)),
       offRampAddress: destChainConfig.offRamp,
-      sender: abi.encodePacked(originalSender),
-      receiver: validateDestChainAddress(message.receiver, destChainConfig.addressBytesLength),
+      sender: abi.encode(originalSender), // Notice only encode to keep backwards compatability.
```

Comment:
Comment is confusing, maybe note something around "source addresses are abi encoded, dest addresses are encoded packed".